libs/langchain/langchain/vectorstores/redis/base.py

```diff
@@ -281,11 +358,62 @@ def similarity_search(
         Returns:
             List[Document]: A list of documents that are most similar to the query text.
         """
-        docs_and_scores = self.similarity_search_with_score(query, k=k)
-        return [doc for doc, _ in docs_and_scores]
+
+        # Creates embedding vector from user query
+        embedding = self.embedding_function(query)
+
+        # Creates Redis query
+        params_dict: Dict[str, Union[str, bytes]] = {
+            "vector": _array_to_buffer(embedding),
+        }
+
+        return_fields = [self._schema.content_key]
+        if return_metadata:
+            return_fields.extend(self._schema.metadata_keys)
+            return_fields.append("score")
+        if return_scores:
+            return_fields.append("score")
+
+        if score_threshold is None:
+            redis_query = self._prepare_vector_query(
+                k, filter=filter, return_fields=return_fields
+            )
+        else:
+            redis_query = self._prepare_range_query(
+                k, filter=filter, return_fields=return_fields
+            )
+            params_dict["score_threshold"] = str(score_threshold)
```

Comment:
```suggestion
        redis_query = self._prepare_vector_query(
            k, filter=filter, return_fields=return_fields
        )
        if score_threshold is not None:
            params_dict["score_threshold"] = str(score_threshold)
```

Reply:
I might be missing something here, but if there is a score threshold, it needs to call _prepare_range_query()

...this wouldn't work? (might be github viewer tbh)